.github/workflows/ci.yaml

```diff
@@ -550,6 +547,11 @@ jobs:
       - name: Check dirty
         run: |
           ./script/check_dirty
+      - name: Minimize uv cache
+        if: steps.cache-uv.conclusion == 'success'
+        run: |
+          . venv/bin/activate
+          uv cache prune --ci
```

Comment:
Did a quick performance comparison as well.
- Current setup: `13s`
- With `cache prune`: `22s`

It doesn't matter much in the grand scheme of things but as mentioned earlier, the goal was to prevent additional downloads which `uv cache prune --ci` doesn't archive.